app/src/main/java/com/keylesspalace/tusky/components/drafts/DraftsActivity.kt

```diff
@@ -0,0 +1,195 @@
+                adapter.submitList(draftList)
+            }
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.drafts, menu)
+        oldDraftsButton = menu.findItem(R.id.action_old_drafts)
+        viewModel.showOldDraftsButton()
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .autoDispose(this, Lifecycle.Event.ON_DESTROY)
+                .subscribe { showOldDraftsButton ->
+                    oldDraftsButton?.isVisible = showOldDraftsButton
+                }
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            android.R.id.home -> {
+                onBackPressed()
+                return true
+            }
+            R.id.action_old_drafts -> {
+                val intent = Intent(this, SavedTootActivity::class.java)
+                startActivityWithSlideInAnimation(intent)
+                return true
+            }
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    override fun onOpenDraft(draft: DraftEntity) {
+
+        if (draft.inReplyToId != null) {
+            bottomSheet.state = BottomSheetBehavior.STATE_COLLAPSED
+            viewModel.getToot(draft.inReplyToId)
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .autoDispose(this)
+                    .subscribe({ status ->
+                        val composeOptions = ComposeActivity.ComposeOptions(
+                                draftId = draft.id,
+                                tootText = draft.content,
+                                contentWarning = draft.contentWarning,
+                                inReplyToId = draft.inReplyToId,
+                                replyingStatusContent = status.content.toString(),
+                                replyingStatusAuthor = status.account.localUsername,
+                                draftAttachments = draft.attachments,
+                                poll = draft.poll,
+                                sensitive = draft.sensitive,
+                                visibility = draft.visibility
+                        )
+
+                        bottomSheet.state = BottomSheetBehavior.STATE_HIDDEN
+
+                        startActivity(ComposeActivity.startIntent(this, composeOptions))
+
+                    }, { throwable ->
+
+                        bottomSheet.state = BottomSheetBehavior.STATE_HIDDEN
+
+                        Log.w(TAG, "failed loading reply information", throwable)
+
+                        if (throwable is HttpException && throwable.code() == 404) {
+                            // the original status to which a reply was drafted has been deleted
+                            // let's open the ComposeActivity without reply information
+                            openDraftWithoutReply(draft)
```

Comment:
It can probably take people by surprise, maybe we should let them know somehow?

Reply:
Very good point